CONTRIBUTING.md

```diff
@@ -44,170 +64,249 @@ not to be able to mint negative values
--  Generally, try to stick to one commit per meaningful change.
-  -  If you fixed several issues in one PR, give them separate commits.
-  -  As mentioned previously changes to the `schema` and the API should be done in appropriate commits separate from the rest of your work.
-  -  Don't bother with separate commits for fixing review comments. Amend the last one, unless the review comment asks to change the `schema`-affecting work. In that case, you want to rebase interactively.
-  -  Tests for functionality in the same commit as the functionality.
+- [Fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo) the [repository](https://github.com/hyperledger/iroha/tree/iroha2-dev) and [create a feature branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-and-deleting-branches-within-your-repository) for your contributions.
+- Use the [Git Rebase Workflow](https://git-rebase.io/). Avoid using `git pull`. Use `git pull --rebase` instead.
+
+Follow these commit guidelines:
+
+- **Sign-off every commit**. If you don't, [DCO](https://github.com/apps/dco) will not let you merge.
+  
+  You may use `git commit -s` to automatically add `Signed-off-by: $NAME <$EMAIL>` as the final line of your commit message.
+
+- Follow the same naming schema as for [pull request titles](#pull-request-titles): `[type] #<issue number>: Description`. This means:
+  - **Use present tense** ("Add feature", not "Added feature")
+  - **Use imperative mood** ("Deploy to docker..." not "Deploys to docker...")
+- Write a meaningful commit message.
+- Try keeping a commit message short.
+- If you need to have a longer commit message:
+  - Limit the first line of your commit message to 50 characters or less.
+  - The first line of your commit message should contain the summary of the work you've done. If you need more than one line, leave a blank line between each paragraph and describe your changes in the middle. The last line must be the sign-off.
+- If you modify the Schema (check by generating the schema with `kagami schema` and diff), you should make all changes to the schema in a separate commit with the message `[schema]`.
+- Try to stick to one commit per meaningful change.
+  - If you fixed several issues in one PR, give them separate commits.
+  - As mentioned previously, changes to the `schema` and the API should be done in appropriate commits separate from the rest of your work.
+  - Don't bother with separate commits for fixing review comments. Amend the last one, unless the review comment asks to change the `schema`-affecting work. In that case, you want to rebase interactively.
+  - Add tests for functionality in the same commit as that functionality.
+
+## Tests and Benchmarks
+
+- To run the source-code based tests, execute [`cargo test`](https://doc.rust-lang.org/cargo/commands/cargo-test.html) in the Iroha root.N Note that this is a long process.
```

Comment:
A typo here? Also could give a ballpark for test runs. E.g. on my 6th gen i7 Intel the full test suite can take upward of 40 minutes. For weaker setups, it might be okayish to just run the tests in CI after creating the PR, as there it's only 20-25 mins AFAICT.